packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -165,21 +171,13 @@ const focusableElements = [
   '[contenteditable]'
 ];
 
-const FOCUSABLE_ELEMENT_SELECTOR = focusableElements.join(',') + ',[tabindex]';
+const FOCUSABLE_ELEMENT_SELECTOR = focusableElements.join(':not([hidden]),') + ',[tabindex]:not([hidden])';
 
 focusableElements.push('[tabindex]:not([tabindex="-1"]):not([disabled])');
-const TABBABLE_ELEMENT_SELECTOR = focusableElements.join(':not([tabindex="-1"]),');
-
-function getFocusableElementsInScope(scope: HTMLElement[], opts: FocusManagerOptions): HTMLElement[] {
-  let res = [];
-  let selector = opts.tabbable ? TABBABLE_ELEMENT_SELECTOR : FOCUSABLE_ELEMENT_SELECTOR;
-  for (let node of scope) {
-    if (node.matches(selector)) {
-      res.push(node);
-    }
-    res.push(...Array.from(node.querySelectorAll(selector)));
-  }
-  return res;
+const TABBABLE_ELEMENT_SELECTOR = focusableElements.join(':not([hidden]):not([tabindex="-1"]),');
+
+function getScopeRoot(scope: HTMLElement[]) {
+  return scope.length ? scope[0].parentElement : null;
```

Comment:
Possibly, I'll have to check. If there are no children in the scope, the `getFocusableTreeWalker` probably shouldn't do anything anyway.

Reply:
@devongovett I think the case of `getFocusableTreeWalker` breaking when `null` is passed as the root node is a moot point, because in each case where `getScopeRoot` is called, the `scope.length` will have a length:

In `focusNext` returned by `createFocusManager`:
https://github.com/adobe/react-spectrum/blob/b9e06919bd772a844edb55d6342696b5940448e5/packages/%40react-aria/focus/src/FocusScope.tsx#L126

In `focusPrevious` returned by `createFocusManager`:
https://github.com/adobe/react-spectrum/blob/b9e06919bd772a844edb55d6342696b5940448e5/packages/%40react-aria/focus/src/FocusScope.tsx#L143

In `onKeyDown` event handler in `useFocusContainment`, where if the `document.activeElement` is not in scope the tree walker will not be created:
https://github.com/adobe/react-spectrum/blob/b9e06919bd772a844edb55d6342696b5940448e5/packages/%40react-aria/focus/src/FocusScope.tsx#L204

And in `focusFirstInScope`:
https://github.com/adobe/react-spectrum/blob/b9e06919bd772a844edb55d6342696b5940448e5/packages/%40react-aria/focus/src/FocusScope.tsx#L297

I may simplify `getScopeRoot` to simply return `scope[0].parentElement`.